PictureApp/Controller/PicturesViewController.swift

```diff
@@ -0,0 +1,306 @@
+//
+//  ViewController.swift
+//  PictureApp
+//
+//  Created by Irina Gorbachenko on 26.11.2021.
+//
+
+import UIKit
+import NetworkLayer
+
+class PicturesViewController: UIViewController, OptionsViewControllerDelegate,FilterViewControllerDelegate {
+    
+    var totalHits = 0
+    var currentPage = 1
+    var selectedCategoryForSort = [String]()
+    var orderForCategories : Order = .popular
+    var collectionViewFlowLayout : UICollectionViewFlowLayout!
+    private let collectionView = UICollectionView (frame: .zero, collectionViewLayout: UICollectionViewFlowLayout())
+    
+    var hits = [Post](){
+        didSet{
+            DispatchQueue.main.async{
+                self.collectionView.reloadData()
+            }
+        }
+    }
+    
+    
+    // MARK:Delegate for categories
+    func filterViewController(_ controller: FilterViewController, didSelectCategory selectedCategory: [String]) {
+         currentPage = 1
+         hits.removeAll()
+         fetchHits(order: orderForCategories, selectedCategory: selectedCategory,currentPage: currentPage)
+          DispatchQueue.main.async{
+            self.collectionView.reloadData()
```

Comment:
Why do we need it here? It will be called before fetchHits method will be executed.

Reply:
Fixed